### PR TITLE
[needsuplift] Fix #430: Fix NSKeyedArchiver/NSKeyedUnarchiver deprecation warnings

### LINKED ISSUE #430: NSKeyedArchiver/NSKeyedUnarchiver deprecation warnings
There are three uses of deprecated APIs in UserReferralProgram.swift. These should be updated to their recommended equivalents.

### PR DESCRIPTION
**Pull request by cndouglas**
This fixes the NSKeyedArchiver/NSKeyedUnarchiver deprecation warnings, as shown in #430.

## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [x] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`

- [x] I have updated the *Unit Tests* to cover new or changed functionality
- [x] I have updated the *UI Tests* to cover new or changed functionality
- [x] I have marked the bug with `[needsuplift]`
- [x] I have made sure that localizable strings use `NSLocalizableString()`

### CODE REVIEW

File: BraveShared/Analytics/UserReferralProgram.swift [hunk 1]
```diff
@@ -50,7 +50,9 @@ public class UserReferralProgram {
             
             if ref.isExtendedUrp() {
                 if let headers = ref.customHeaders {
-                    Preferences.URP.customHeaderData.value = NSKeyedArchiver.archivedData(withRootObject: headers)
+                    do {
+                        try Preferences.URP.customHeaderData.value = NSKeyedArchiver.archivedData(withRootObject: headers, requiringSecureCoding: false)
+                    } catch {}
```
Comment (kylehickinson) [inline BraveShared/Analytics/UserReferralProgram.swift]: Can you add an error here?

```suggestion
                    } catch {
                        log.error("Failed to save URP custom header data \(headers) with error: \(error.localizedDescription)")
                    }
```

File: BraveShared/Analytics/UserReferralProgram.swift [hunk 2]
```diff
@@ -178,14 +180,16 @@ public class UserReferralProgram {
         service.fetchCustomHeaders() { headers, error in
             if headers.isEmpty { return }
             
-            Preferences.URP.customHeaderData.value = NSKeyedArchiver.archivedData(withRootObject: headers)
+            do {
+                try Preferences.URP.customHeaderData.value = NSKeyedArchiver.archivedData(withRootObject: headers, requiringSecureCoding: false)
+            } catch {}
```
Comment (kylehickinson) [inline BraveShared/Analytics/UserReferralProgram.swift]: Same as above

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Thanks for grabbing this :)

File: BraveShared/Analytics/UserReferralProgram.swift (hunk 1 above)
Comment (cndouglas) [reply BraveShared/Analytics/UserReferralProgram.swift]: Will do.

**Comment by cndouglas** [comment]
@kylehickinson I added error handling as requested.

**Review by kylehickinson — APPROVED** [review]

**Comment by kylehickinson** [comment]
Thanks again @Liunkae!

**Pull request merged**